core/crypto/hash/low_level.odin

```diff
@@ -0,0 +1,287 @@
+package crypto_hash
+
+import "core:crypto/sha2"
+import "core:crypto/sha3"
+import "core:crypto/shake"
+import "core:crypto/sm3"
+
+import "core:crypto/legacy/md5"
+import "core:crypto/legacy/sha1"
+
+// Algorithm is the algorithm identifier associated with a given Context.
+Algorithm :: enum {
+	Invalid,
+	SHA224,
+	SHA256,
+	SHA384,
+	SHA512,
+	SHA512_256,
+	SHA3_224,
+	SHA3_256,
+	SHA3_384,
+	SHA3_512,
+	SHAKE_128,
+	SHAKE_256,
+	SM3,
+	Insecure_MD5,
+	Insecure_SHA1,
+}
+
+// ALGORITHM_NAMES is the Algorithm to algorithm name string map.
+ALGORITHM_NAMES := map[Algorithm]string {
+	.Invalid       = "Invalid",
+	.SHA224        = "SHA-224",
+	.SHA256        = "SHA-256",
+	.SHA384        = "SHA-384",
+	.SHA512        = "SHA-512",
+	.SHA512_256    = "SHA-512/256",
+	.SHA3_224      = "SHA3-224",
+	.SHA3_256      = "SHA3-256",
+	.SHA3_384      = "SHA3-384",
+	.SHA3_512      = "SHA3-512",
+	.SHAKE_128     = "SHAKE-128",
+	.SHAKE_256     = "SHAKE-256",
+	.SM3           = "SM3",
+	.Insecure_MD5  = "MD5",
+	.Insecure_SHA1 = "SHA-1",
+}
+
+// DIGEST_SIZES is the Algorithm to digest size map.
+DIGEST_SIZES := map[Algorithm]int {
```

Comment:
Likewise seems like this could be `DIGEST_SIZES := [Algorithm]int{ ... }` instead of a `map`.